### PR TITLE
Check the latest version from Google Cloud Storage

### PR DESCRIPTION
**Pull request by pepibumur**
Related https://github.com/tuist/tuist/issues/848

### Short description 📝
As explained [in this issue](https://github.com/tuist/tuist/issues/848) we are moving away from GitHub releases to Google Cloud Storage as a place to store our releases. Several users ran into issues in their projects because they were hitting some limits on the GitHub API.

### Solution 📦
This is the first of 2 PRs to make this change possible. It adds a new utility, `GoogleCloudStorageClient` to interact with Google Cloud Storage, and uses it from the `Updater` utility to check what's the latest version available.

In a follow-up PR, I'll extend `GoogleCloudStorageClient` to support downloading a given version and update the `Updater` accordingly. After that users should no longer run into API limit issues.

### How to test
Check out the branch and run:

```
rm -rf ~/.tuist/Versions
swift run tuistenv update
```

tuistenv should be able to determine the latest version and update to it.

### CODE REVIEW

File: Sources/TuistSupport/Utils/URLSessionScheduler.swift [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    private let requestTimeout: Double
+
+    /// Initializes the client with the session.
+    ///
+    /// - Parameter session: url session.
+    /// - Parameter requestTimeout: request timeout.
+    public init(session: URLSession = URLSession.shared,
+                requestTimeout: Double = URLSessionScheduler.defaultRequestTimeout) {
+        self.session = session
+        self.requestTimeout = requestTimeout
```
Comment (kwridan) [inline Sources/TuistSupport/Utils/URLSessionScheduler.swift]: I believe this needs to be set on a `URLSessionConfiguration`
https://developer.apple.com/documentation/foundation/urlsessionconfiguration

File: Sources/TuistSupport/Utils/URLSessionScheduler.swift [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    public func schedule(request: URLRequest) -> (error: Error?, data: Data?) {
+        var data: Data?
+        var error: Error?
+        let semaphore = DispatchSemaphore(value: 0)
+        session.dataTask(with: request) { sessionData, _, sessionError in
+            data = sessionData
+            error = sessionError
+            semaphore.signal()
+        }.resume()
+        _ = semaphore.wait(timeout: .now() + 3)
```
Comment (kwridan) [inline Sources/TuistSupport/Utils/URLSessionScheduler.swift]: was the intention to also use the timeout here?

File: Sources/TuistSupport/Utils/URLSessionScheduler.swift (hunk 1 above)
Comment (pepibumur) [reply Sources/TuistSupport/Utils/URLSessionScheduler.swift]: Good catch. I updated it accordingly.

File: Sources/TuistSupport/Utils/URLSessionScheduler.swift (hunk 2 above)
Comment (pepibumur) [reply Sources/TuistSupport/Utils/URLSessionScheduler.swift]: Right. I ended up removing it from here and using URLSessionConfiguration as you suggested.

**Pull request merged**